src/core/public/http/fetch.ts

```diff
@@ -125,7 +127,7 @@ export class Fetch {
         'Content-Type': 'application/json',
         ...options.headers,
         'kbn-version': this.params.kibanaVersion,
-        ...(options.context ? new ExecutionContextContainer(options.context).toHeader() : {}),
+        ...(!isEmpty(context) ? new ExecutionContextContainer(context).toHeader() : {}),
```

Comment:
Under what circumstances can `context` be an empty object?

Reply:
For example if requests are being set before the `application.currentApp$` hasn't been initialized yet.
but normally it shouldn't happen, no